application.go

```diff
@@ -36,23 +37,23 @@ type Applications struct {
 // Application is the definition for an application in marathon
 type Application struct {
 	ID                    string              `json:"id,omitempty"`
-	Cmd                   string              `json:"cmd,omitempty"`
-	Args                  []string            `json:"args"`
-	Constraints           [][]string          `json:"constraints"`
+	Cmd                   *string             `json:"cmd,omitempty"`
+	Args                  *[]string           `json:"args,omitempty"`
```

Comment:
If by blank you mean empty string, then `/bin/sh [-c] ""` makes sense because the

> [cmd] value is wrapped by Mesos via `/bin/sh -c ${app.cmd}`.

(From the [POST docs](https://mesosphere.github.io/marathon/docs/rest-api.html#post-v2-apps).)

Not very helpful from a semantical point of view, but syntactically correct I guess. :-)

We run Docker containers in our Marathon/Mesos cluster exclusively, and more often than not leave both `cmd` and `args` empty. So I'm not sure what to make out of the Marathon documentation saying that _either `cmd` or `args` must be supplied_. Maybe it's set implicitly when using a Docker definition.

In that sense, I'm fine with your approach here.